Award player for map exploration

AlexanderOlkhovskiy opened the pull request:
Award is given based on percentage of explored map

AlexanderOlkhovskiy committed:
Award player for map exploration

vladislavbelov commented on simulation/components/MapExplorationTracker.js:
```diff
@@ -0,0 +1,67 @@
+{
+	let newPercentMapExplored = this.GetPercentMapExplored();
+	let percentMapExploredDiff = newPercentMapExplored - this.percentMapExplored;
+	if (percentMapExploredDiff > 0)
```
Early return?

AlexanderOlkhovskiy replied:
Done in ebf8499

vladislavbelov commented on simulation/components/MapExplorationTracker.js:
```diff
@@ -0,0 +1,67 @@
+	let awardMessage = "Map exploration award: ";
+	let firstResource = true;
+	for (let type of Resources.GetCodes())
+		if (award[type] > 0)
```
```
let awards = [];
for (let type of Resources.GetCodes())
    if (award[type] > 0)
        awards.push("" + award[type] + " " + type);
awardMessage += ", ".join(awards);
```

AlexanderOlkhovskiy replied:
Done in add4bf0

vladislavbelov commented on simulation/components/MapExplorationTracker.js:
```diff
@@ -0,0 +1,67 @@
+			awardMessage += "" + award[type] + " " + type;
+			firstResource = false;
+		}
+	let cmpGUIInterface = Engine.QueryInterface(SYSTEM_ENTITY, IID_GuiInterface);
```
Should be here a component check? Because GUI interface may not exist in non-visual replays, no?

AlexanderOlkhovskiy replied:
There are no checks in other places, an example: https://github.com/0ad/0ad/blob/master/binaries/data/mods/public/simulation/components/Player.js#L308
I guess in the non-visual mode the component itself is still present, just doing nothing or something like that.

AlexanderOlkhovskiy committed:
MapExplorationTracker refactoring: early return

As suggested by vladislavbelov in https://github.com/AlexanderOlkhovskiy/0ad-space/pull/1

AlexanderOlkhovskiy committed:
MapExplorationTracker refactoring: join resource descriptions via Array.prototype.join()

As suggested by vladislavbelov in https://github.com/AlexanderOlkhovskiy/0ad-space/pull/1
